packages/teleterm/src/services/config/types.ts

```diff
@@ -0,0 +1,18 @@
+import { Platform } from '../../mainProcess/types';
+import { KeyboardShortcutsConfig } from './providers/keyboardShortcutsConfigProvider';
+
+export interface Config {
+  keyboardShortcuts: KeyboardShortcutsConfig;
+}
+
+export interface ConfigServiceProvider<
+  T extends Record<string, number | string | boolean>
+  > {
+  provideDefaults?(): Partial<T>;
+  provideDefaultsForPlatform?(platform: Platform): Partial<T>;
```

Comment:
combine them in 1 method?

Reply:
I did it this way because at the same time we can specify some of the defaults for all platforms and some of them only for a single platform. But I can agree, they can be combined into a 1 method, it will be more type-safe as we get rid of `Partials`